### DISCUSSION TITLE
AOC Day 11 Discussion -  Monkey in the Middle 🙈

### DISCUSSION THREAD

**Discussion by doleraj** [Virtual-Coffee/virtualcoffee.io#770] [Advent of Code 2022]
Discussion of solutions for [Advent of Code Day 11](https://adventofcode.com/2022/day/11) here.

---
Don't forget about the VC Advent of Code private leaderboard:

Go to [leaderboard/private](https://adventofcode.com/2022/leaderboard/private)
Use join code 363496-e6f42283

**Comment by doleraj** [comment]
I am grateful to my Discrete and Combinatorics teacher for managing to get knowledge about Rings to sink into my skull. Couldn't have actually solved this without that. https://github.com/doleraj/adventofcode/blob/main/aoc2022/src/day11/index.ts

**Comment by rhdeck** [comment]
Part one came together pretty quickly, part 2 took more thinking and research. For me it was a math problem about the property of modules - but got to the same place as Arthur! [solve](https://github.com/rhdeck/aoc-2022/blob/main/src/11/day_11.ipynb)

**Comment by davidalpert** [comment]
[Part 1](https://github.com/davidalpert/advent-of-code-da/blob/a25bd2f3e785b243e3afe4025fa910bcbe069c32/2022/day11_monkey_in_the_middle.fs) came quickly to me also, and Part 2 threw me for a loop until a hint from @tkshill that there was a math trick involved prompted me to reread the instructions and figure out why [my Part 2 example test results were diverging after lots of rounds](https://github.com/davidalpert/advent-of-code-da/blob/21bd6c65c700d87e8385ceddf107118fc93511e6/2022/day11_tests.fs#L194-L198) (that test failed while the tests after rounds 1 and 20 passed).

That's when I realized the difference between what I was tracking throughout each round and [what I actually needed to track](https://github.com/davidalpert/advent-of-code-da/blob/2022/2022/day11_monkey_in_the_middle.fs#L105-L109) based on the math involved:

- https://github.com/davidalpert/advent-of-code-da/blob/2022/2022/day11_monkey_in_the_middle.fs

**Comment by tkshill** [comment]
This was **rough** for me. Mad it took me so long, for no reason.

[solve](https://github.com/tkshill/AdventOfCode2022/blob/main/AOC2022/Day11.fs)
